src/main/java/com/laytonsmith/core/functions/DataHandling.java

```diff
@@ -2040,6 +2353,54 @@ public Mixed execs(Target t, Environment env, Script parent, ParseTree... nodes)
 			return closure;
 		}
 
+		@Override
+		public Scope linkScope(StaticAnalysis analysis, Scope parentScope, ParseTree ast,
+				Environment env, Set<ConfigCompileException> exceptions) {
+			return this.linkScope(analysis, parentScope, ast, env, exceptions, true);
+		}
+
+		public Scope linkScope(StaticAnalysis analysis, Scope parentScope, ParseTree ast,
+				Environment env, Set<ConfigCompileException> exceptions, boolean codeInheritsParentScope) {
+
+			// Handle empty closure.
+			if(ast.numberOfChildren() == 0) {
+				return parentScope;
+			}
+
+			// Handle optional return type argument.
+			int ind = 0;
+			Scope retTypeScope = (ast.getChildAt(ind).getData().isInstanceOf(CClassType.TYPE)
```

Comment:
Some code could be removed there. That term should always be a CClassType and otherwise generate an error elsewhere (could be during typechecking, but the rewrite step after parsing might be better, if we consider this a syntax error). In case it is not a CClassType, someone could have any code here. Not using this scope just means that whatever you declare in that term won't be usable after that term.